src/simplify_qdq.cpp

```diff
@@ -0,0 +1,156 @@
+#include <migraphx/simplify_qdq.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/shape.hpp>
+#include <migraphx/matcher.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/op/convolution.hpp>
+#include <migraphx/op/quant_convolution.hpp>
+#include <migraphx/op/dot.hpp>
+#include <migraphx/op/quant_dot.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+std::unordered_set<std::string> get_quantizable_op_names()
+{
+    static std::unordered_set<std::string> s = {"convolution", "dot"};
+    return s;
+}
+
+struct match_find_add_bias
+{
+    auto matcher() const
+    {
+        return match::name("dequantizelinear")(match::arg(0)(
+            match::name("quantizelinear")(
+                match::arg(0)(match::name("add")(
+                                  match::arg(0)(match::name("convolution").bind("convolution")))
+                                  .bind("add")))
+                .bind("quantizelinear")));
+    }
+
+    void apply(module& m, match::matcher_result r) const
+    {
+        auto dq   = r.result;
+        auto q    = r.instructions["quantizelinear"];
+        auto add  = r.instructions["add"];
+        auto conv = r.instructions["convolution"];
+
+        auto q_args = q->inputs();
+        for(auto&& arg : q_args)
+        {
+            if(arg->name() == "multibroadcast" or arg->name() == "broadcast")
+            {
+                arg = m.insert_instruction(add, arg->get_operator(), arg->inputs());
+            }
+        }
+        m.replace_instruction(q, q->get_operator(), q_args);
+        q_args.front() = conv;
+        auto new_q     = m.insert_instruction(add, migraphx::make_op("quantizelinear"), q_args);
+        auto dq_args   = dq->inputs();
+        q_args.front() = dq_args.front();
+        m.replace_instruction(dq, dq->get_operator(), q_args);
+        q_args.front()   = new_q;
+        auto new_dq      = m.insert_instruction(add, migraphx::make_op("dequantizelinear"), q_args);
+        auto add_args    = add->inputs();
+        add_args.front() = new_dq;
+        m.replace_instruction(add, migraphx::make_op("add"), add_args);
+    }
+};
+
+struct match_find_quantizable_ops
+{
+    auto matcher() const
+    {
+        return match::name("dequantizelinear")(match::arg(0)(match::name("quantizelinear")(
+            match::arg(0)(match::name(get_quantizable_op_names()).bind("qop")))));
```

Comment:
This is also assumes there are QDQ after the operator, which is not true. According to our previous discussion, and what ORT mentioned before, we can only assume there is QDQ for each argument of the operators to be quantized. 

Because of this reason, for the alexnet model I shared with you, there are two convolution ops that are not converted to quant_convolution. You can run the model for a try.